.github/actions/main-sample-app-build/action.yml

```diff
@@ -0,0 +1,69 @@
+name: Get or save latest main branch sample app build
+description: Use to either save or get the latest sample app build for the APN sample app, from the main branch. The build is used to generate the SDK size diff reports. 
+
+inputs:
+  # If you want to save a new build, pass in this input. If you want to get the latest build, leave this input out.
+  set-latest-main-build:
+    description: 'The name of the xcarchive for the APN sample app to save. Example: "App.xcarchive"'
+    type: string
+    required: false 
+    default: ''
+
+outputs:
+  apn-app-xcarchive-path:
+    description: 'The full relative path to the xcarchive for the APN sample app. Example: "Apps/APN-UIKit/build/MainBranchApp.xcarchive"'
+    value: ${{ steps.set-action-outputs.outputs.apn-app-xcarchive-path }}
+  apn-app-xcarchive-name:
+    description: 'The name of the xcarchive for the APN sample app. Example: "MainBranchApp.xcarchive"'
+    value: ${{ steps.set-action-outputs.outputs.apn-app-xcarchive-name }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setting environment variables for convenience in future steps 
+      shell: bash
+      # Use each of the set variables in future steps like: ${{ env.APN_SAMPLE_APP_BUILD_PATH }}      
+      run: |
+        echo "APN_SAMPLE_APP_BUILD_PATH=Apps/APN-UIKit/build" >> $GITHUB_ENV
```

Comment:
What are the possible values of `GITHUB_ENV` ?

Reply:
GITHUB_ENV contains all environment variables of the CI job. So, the possibilities are infinite! 

The syntax of this line of code, `echo "FOO=BAR" >> $GITHUB_ENV`, is how you set new environment variables on github actions. I understand that might not be very clear looking at it.